### PR TITLE
fix: removing surplus slash, making logs richer

### PR DESCRIPTION
**Pull request by javanlacerda**
#### Summary
During the validations, we noticed that the the template for `source-repository-ref` for gitlab has a surplus slash.
We also noticed that the error with log `"value <> not present in either claims or defaults.` has spawned several times in production, but we wasn't able to detect the reason for it, then I decided to put at least the issuer that triggered the flow on the error log.
I also added a validation that the metadata should exist for ci-providers.

#### Documentation

cc @haydentherapper

### CODE REVIEW

File: pkg/identity/ciprovider/principal.go [hunk 1]
```diff
@@ -135,10 +143,10 @@ func (principal ciPrincipal) Embed(_ context.Context, cert *x509.Certificate) er
 		s := v.Field(i).String() // value of each field, e.g the template string
 		// We check the field name to avoid to apply the template for the Issuer
 		// Issuer field should always come from the token issuer
-		if s == "" || vType.Field(i).Name == "Issuer" {
+		if strings.Trim(s, " ") == "" || vType.Field(i).Name == "Issuer" {
```
Comment (haydentherapper) [inline pkg/identity/ciprovider/principal.go:146]: Can you use `TrimSpace` instead, which will trim any whitespace characters as well?

File: pkg/identity/ciprovider/principal.go [hunk 2]
```diff
@@ -115,7 +120,10 @@ func (principal ciPrincipal) Embed(_ context.Context, cert *x509.Certificate) er
 	if err != nil {
 		return err
 	}
-	subjectAlternativeName, err := applyTemplateOrReplace(principal.ClaimsMetadata.SubjectAlternativeNameTemplate, claims, defaults)
+	if strings.Trim(principal.ClaimsMetadata.SubjectAlternativeNameTemplate, " ") == "" {
```
Comment (haydentherapper) [inline pkg/identity/ciprovider/principal.go:123]: Same here, `TrimSpace`?

File: pkg/identity/ciprovider/principal.go [hunk 3]
```diff
@@ -81,7 +82,7 @@ func applyTemplateOrReplace(extValueTemplate string, tokenClaims map[string]stri
 	}
 	claimValue, ok := mergedData[extValueTemplate]
 	if !ok {
-		return "", fmt.Errorf("value <%s> not present in either claims or defaults", extValueTemplate)
+		return "", fmt.Errorf("value <%s> not present in either claims or defaults. Issuer: %s", extValueTemplate, issuer)
```
Comment (haydentherapper) [inline pkg/identity/ciprovider/principal.go:85]: Any more logging we can do? What about which extension we're trying to write to?

Comment (javanlacerda) [reply pkg/identity/ciprovider/principal.go:85]: I was worried about logging any sensitive information, but I can add the extension for sure. wdyt?

Comment (haydentherapper) [reply pkg/identity/ciprovider/principal.go:85]: Yea, the name of the ext seems fine. It'll help us reproduce this once we know which issuer and which claim is causing the issue.

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**